src/masternode.h

```diff
@@ -258,6 +259,19 @@ class CMasternode
 
     bool IsWatchdogExpired() { return nActiveState == MASTERNODE_WATCHDOG_EXPIRED; }
 
+    bool IsValidForPayment()
+    {
+        if(nActiveState == MASTERNODE_ENABLED) {
+            return true;
+        }
+        if(!sporkManager.IsSporkActive(SPORK_14_REQUIRE_SENTINEL_FLAG) &&
```

Comment:
Maybe I'm blind but I don't see the problem here.

If the masternode is ENABLED we return true, regardless of the spork setting.

Otherwise the only case where we should return true is when the state is WATCHDOG_EXPIRED **and** SPORK_14_REQUIRE_SENTINEL_FLAG is **not** active.

Because if the spork is active then we require the ENABLED state and at this point we have already checked for that.